braindecode/models/sparcnet.py

```diff
@@ -0,0 +1,353 @@
+        (self.add_module("elu2", nn.ELU()),)
+        (
+            self.add_module(
+                "conv2",
+                nn.Conv1d(
+                    bn_size * growth_rate,
+                    growth_rate,
+                    kernel_size=3,
+                    stride=1,
+                    padding=1,
+                    bias=conv_bias,
+                ),
+            ),
+        )
+        self.drop_rate = drop_rate
+
+    def forward(self, x):
+        new_features = super(DenseLayer, self).forward(x)
+        new_features = F.dropout(new_features, p=self.drop_rate, training=self.training)
+        return torch.cat([x, new_features], 1)
+
+
+class DenseBlock(nn.Sequential):
+    """
+    A densely connected block that uses DenseLayers.
+
+    Parameters
+    ----------
+    num_layers : int
+        Number of layers in this block.
+    input_channels : int
+        Number of input channels.
+    growth_rate : int
+        Rate of growth of channels in this layer.
+    bn_size : int
+        Multiplicative factor for the bottleneck layer (does not affect the output size).
+    drop_rate : float, optional
+        Dropout rate. Default is 0.5.
+    conv_bias : bool, optional
+        Whether to use bias in convolutional layers. Default is True.
+    batch_norm : bool, optional
+        Whether to use batch normalization. Default is True.
+
+    Examples
+    --------
+    >>> x = torch.randn(128, 5, 1000)
+    >>> batch, channels, length = x.shape
+    >>> model = DenseBlock(3, channels, 5, 2)
+    >>> y = model(x)
+    >>> y.shape
+    torch.Size([128, 20, 1000])
+    """
+
+    def __init__(
+        self,
+        num_layers,
+        input_channels,
+        growth_rate,
+        bn_size,
+        drop_rate=0.5,
+        conv_bias=True,
+        batch_norm=True,
+    ):
+        super(DenseBlock, self).__init__()
+        for idx_layer in range(num_layers):
+            layer = DenseLayer(
+                input_channels + idx_layer * growth_rate,
+                growth_rate,
+                bn_size,
+                drop_rate,
+                conv_bias,
+                batch_norm,
+            )
+            self.add_module(f"denselayer{idx_layer + 1}", layer)
+
+
+class TransitionLayer(nn.Sequential):
+    """
+    A pooling transition layer.
+
+    Parameters
+    ----------
+    input_channels : int
+        Number of input channels.
+    output_channels : int
+        Number of output channels.
+    conv_bias : bool, optional
+        Whether to use bias in convolutional layers. Default is True.
+    batch_norm : bool, optional
+        Whether to use batch normalization. Default is True.
+
+    Examples
+    --------
+    >>> x = torch.randn(128, 5, 1000)
+    >>> model = TransitionLayer(5, 18)
+    >>> y = model(x)
+    >>> y.shape
+    torch.Size([128, 18, 500])
+    """
+
+    def __init__(
+        self, input_channels, output_channels, conv_bias=True, batch_norm=True
```

Comment:
you should aim for consistency with pytorch naming conventions as much as possible so there use in_channels, out_channels